default/scripting/techs/ship_weapons/ship_weapons.py

```diff
@@ -1,39 +1,67 @@
 from common.misc import SHIP_WEAPON_DAMAGE_FACTOR
+from common.priorities import AFTER_ALL_TARGET_MAX_METERS_PRIORITY, DEFAULT_PRIORITY
 from techs.techs import (
     ARBITRARY_BIG_NUMBER_FOR_METER_TOPUP,
     EMPIRE_OWNED_SHIP_WITH_PART,
     SHIP_PART_UPGRADE_RESUPPLY_CHECK,
 )
 
 
+# Tech based setting of damage and number of shots of a weapon. Registers named reals for use in the pedia.
+# Be careful the effect does not show in the ship damage/shot estimates in the ship designer UI if the tech is not researched yet.
 # @1@ part name
 def WEAPON_BASE_EFFECTS(part_name: str):
     # these NamedReals need to be parsed to be registered in the pedia. XXX remove this when the pedia can look up ship meters/part capacity
     NamedReal(name=part_name + "_PART_CAPACITY", value=PartCapacity(name=part_name))
     NamedReal(name=part_name + "_PART_SECONDARY_STAT", value=PartSecondaryStat(name=part_name))
 
-    # The following is really only needed on the first resupplied turn after an upgrade is researched, since the resupply currently
-    # takes place in a portion of the turn before meters are updated, but currently there is no good way to restrict this to
-    # only that first resupply (and it is simply mildly inefficient to repeat the topup later).
-    topup_effects_group = EffectsGroup(
-        scope=EMPIRE_OWNED_SHIP_WITH_PART(part_name) & Turn(high=LocalCandidate.LastTurnResupplied),
+    # Set the max meters for damage and number of shots to the base value from the ship part specification.
+    # The default effects of a direct weapon part sets the capacity to a scaled value
+    # The default effects of a fighter hangar part sets the secondary stat to a scaled value
+    # Note: So direct weapon parts with NoDefaultCapacityEffect will not have the damage/PartCapacity scaled by SHIP_DAMAGE_WEAPON_FACTOR.
+    set_max_meters_effects_group = EffectsGroup(
+        scope=EMPIRE_OWNED_SHIP_WITH_PART(part_name),
         accountinglabel=part_name,
-        effects=SetCapacity(partname=part_name, value=Value + ARBITRARY_BIG_NUMBER_FOR_METER_TOPUP),
+        priority=DEFAULT_PRIORITY,
+        effects=[
+            SetMaxCapacity(partname=part_name, value=PartCapacity(name=part_name)),
+            SetMaxSecondaryStat(partname=part_name, value=PartSecondaryStat(name=part_name)),
+        ],
     )
-    return [topup_effects_group]
 
+    # Set the damage and shot meters to the max values every turn.
+    # The topup_effects_group is necessary..
+    # 1) the first turn after an upgrade is researched, since the resupply currently takes place in a portion of the turn before meters are updated
```

Comment:
maybe explain a bit more: what "the resupply" taking place means and why that's relevant?